stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/UnsupportedAttachmentFactory.kt

```diff
@@ -0,0 +1,126 @@
+     *
+     * @param message The message containing custom attachments that we are going to render.
+     * @return True it the message contains unsupported attachment.
+     */
+    override fun canHandle(message: Message): Boolean {
+        return message.attachments.isNotEmpty() && message.attachments.all { !it.isSupported() }
+    }
+
+    /**
+     * Creates fallback UI that represents unsupported attachments.
+     *
+     * @param message The message containing custom attachments that we are going to render.
+     * @param listeners [MessageListListenerContainer] with listeners for the message list.
+     * @param parent The parent View where the attachment content view is supposed to be placed.
+     * @return An inner ViewHolder with the fallback attachment content view.
+     */
+    override fun createViewHolder(
+        message: Message,
+        listeners: MessageListListenerContainer?,
+        parent: ViewGroup,
+    ): InnerAttachmentViewHolder {
+        return object : InnerAttachmentViewHolder(UnsupportedAttachmentsView(parent.context)) {}
+    }
+
+    /**
+     * View used to display unsupported attachments.
+     */
+    private class UnsupportedAttachmentsView : FrameLayout {
+
+        private val binding = StreamUiUnsupportedAttachmentViewBinding.inflate(streamThemeInflater, this, true)
+
+        private lateinit var style: UnsupportedAttachmentViewStyle
+
+        constructor(context: Context) : this(context, null, 0)
+
+        constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+
+        constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+            context.createStreamThemeWrapper(),
+            attrs,
+            defStyleAttr
+        ) {
+            init(attrs)
+        }
+
+        private fun init(attrs: AttributeSet?) {
+            style = UnsupportedAttachmentViewStyle(context, attrs)
+
+            val shapeAppearanceModel = ShapeAppearanceModel.Builder()
+                .setAllCorners(CornerFamily.ROUNDED, style.cornerRadius.toFloat())
+                .build()
+            binding.attachmentContainer.background = MaterialShapeDrawable(shapeAppearanceModel).apply {
+                fillColor = ColorStateList.valueOf(style.backgroundColor)
+                strokeColor = ColorStateList.valueOf(style.strokeColor)
+                strokeWidth = style.strokeWidth.toFloat()
+            }
+            binding.titleImageView.setTextStyle(style.titleTextStyle)
+        }
+    }
+
+    /**
+     * Checks if the attachment type is supported.
+     *
+     * @return True if the attachment type is supported.
+     */
+    private fun Attachment.isSupported(): Boolean {
+        return SUPPORTED_ATTACHMENT_TYPES.contains(type)
+    }
+
+    private companion object {
+        /**
+         * The list of supported attachment types.
+         */
+        private val SUPPORTED_ATTACHMENT_TYPES: Set<String> = setOf(
+            ModelType.attach_image,
```

Comment:
What happens if someone adds a custom attachment, do they need to manually alter this list in order to avoid getting unsupported attachment?

Reply:
Tested by generating a local build and plugging it into our custom attachments sample.
Works well without any additional effort. :smile: